writer_downgrade_test.go

```diff
@@ -253,6 +316,23 @@ func (s *WriterDowngradeSuite) validateRead(b *bytes.Buffer) {
 		s.Assert().Nil(err)
 		s.Assert().Equal(15.44, math.Round(price*100)/100)
 
+		// Read the second array element's "variations" field
+		err = r.AdvanceTo(buf, "products", "variations")
+		if err != ErrNoSuchField {
```

Comment:
Why do the tests check for `err != ErrNoSuchField`? Wouldn't we want to fail if the error isn't `nil`? Or are you trying to future proof as the manifest changes? If so, do you think there is a different way to handle that? It feels like this could lead to not catching bugs, but maybe I'm not understanding what this is for.

Reply:
This is a downgrade test, which means that we're using code that expects a newer RSF version, but then also testing the same code against an older RSF version that is missing some fields. 

This could eventually be used in PPM for cases where we've updated our RSF files, but we still want to accommodate air-gapped users who haven't updated their local manifest with the most recent version. Since there are some cases where it's critical to have new data and other cases where missing data can be ignored, it'll have to be on a case-by-case basis very specific to each new field.

Reply:
Okay, that makes more sense. It would be nice modify the tests to test each version individually. So for the test with the old version, it can do `s.Assert().ErrorContains(err, ErrNoSuchField.Error())`. Does that make sense to try to do or not really? I'm okay either way if you don't think it is worth it.

Reply:
That makes sense, I'm fine with that approach!